compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp

```diff
@@ -116,6 +120,22 @@ TEST(TestRadsanInterceptors, posixMemalignDiesWhenRealtime) {
   expectNonrealtimeSurvival(func);
 }
 
+#if SANITIZER_INTERCEPT_MEMALIGN
+TEST(TestRadsanInterceptors, memalignDiesWhenRealtime) {
+  auto func = []() { EXPECT_NE(memalign(2, 2048), nullptr); };
+  expectRealtimeDeath(func, "memalign");
+  expectNonrealtimeSurvival(func);
+}
+#endif
+
+#if SANITIZER_INTERCEPT_PVALLOC
+TEST(TestRadsanInterceptors, pvallocDiesWhenRealtime) {
+  auto func = []() { EXPECT_NE(pvalloc(2048), nullptr); };
+  expectRealtimeDeath(func, "pvalloc");
+  expectNonrealtimeSurvival(func);
```

Comment:
Do we have a general sense of what we want the "full compilation test" vs these unit tests? When is it necessary to have a full lit test vs just having these unit tests?

May be good to standardize around when we think both are necessary. (this is outside of the scope of this review, and does not block it)

Reply:
Good question - I think this warrants a proper discussion. Maximising both test coverage and maintainability is a delicate art form! I do have an opinion that is best expressed in the form of some analogies - but it'll be easiest to explain on a call. Let's schedule a time!

Reply:
I'll add it to the list for our next chat :)

Reply:
Awesome. Looking forward to hearing your opinions too of course!